internal/gps/source_manager_test.go

```diff
@@ -13,71 +13,82 @@ import (
 
 func TestSourceManager_InferConstraint(t *testing.T) {
 	t.Parallel()
-	h := test.NewHelper(t)
-	cacheDir := "gps-repocache"
-	h.TempDir(cacheDir)
-	sm, err := NewSourceManager(h.Path(cacheDir))
-	h.Must(err)
-
-	sv, err := NewSemverConstraintIC("v0.8.1")
-	if err != nil {
-		t.Fatal(err)
-	}
 
-	svs, err := NewSemverConstraintIC("v0.12.0-12-de4dcafe0")
-	if err != nil {
-		t.Fatal(err)
-	}
+	testcase := func(str string, pi ProjectIdentifier, want Constraint) func(*testing.T) {
+		return func(t *testing.T) {
+			t.Parallel()
+			h := test.NewHelper(t)
+			defer h.Cleanup()
 
-	constraints := map[string]Constraint{
-		"":       Any(),
-		"v0.8.1": sv,
-		"v2":     NewBranch("v2"),
-		"v0.12.0-12-de4dcafe0": svs,
-		"master":               NewBranch("master"),
-		"5b3352dc16517996fb951394bcbbe913a2a616e3": Revision("5b3352dc16517996fb951394bcbbe913a2a616e3"),
-
-		// valid bzr rev
-		"jess@linux.com-20161116211307-wiuilyamo9ian0m7": Revision("jess@linux.com-20161116211307-wiuilyamo9ian0m7"),
-		// invalid bzr rev
-		"go4@golang.org-sadfasdf-": NewVersion("go4@golang.org-sadfasdf-"),
-	}
+			cacheDir := "gps-repocache"
+			h.TempDir(cacheDir)
 
-	pi := ProjectIdentifier{ProjectRoot: "github.com/carolynvs/deptest"}
-	for str, want := range constraints {
-		got, err := sm.InferConstraint(str, pi)
-		h.Must(err)
+			sm, err := NewSourceManager(h.Path(cacheDir))
+			h.Must(err)
 
-		wantT := reflect.TypeOf(want)
-		gotT := reflect.TypeOf(got)
-		if wantT != gotT {
-			t.Errorf("expected type: %s, got %s, for input %s", wantT, gotT, str)
-		}
-		if got.String() != want.String() {
-			t.Errorf("expected value: %s, got %s for input %s", want, got, str)
+			got, err := sm.InferConstraint(str, pi)
+			h.Must(err)
+
+			wantT := reflect.TypeOf(want)
+			gotT := reflect.TypeOf(got)
+			if wantT != gotT {
+				t.Errorf("expected type: %s, got %s, for input %s", wantT, gotT, str)
+			}
+			if got.String() != want.String() {
+				t.Errorf("expected value: %s, got %s for input %s", want, got, str)
+			}
 		}
 	}
-}
 
-func TestSourceManager_InferConstraint_InvalidInput(t *testing.T) {
-	h := test.NewHelper(t)
+	var (
+		gitProj = ProjectIdentifier{ProjectRoot: "github.com/carolynvs/deptest"}
+		bzrProj = ProjectIdentifier{ProjectRoot: "launchpad.net/govcstestbzrrepo"}
+		hgProj  = ProjectIdentifier{ProjectRoot: "bitbucket.org/golang-dep/dep-test"}
```

Comment:
Let's add a testcase for svn. Here's the test repo info: https://github.com/golang/dep/blob/master/internal/gps/vcs_repo_test.go#L122

Reply:
Will do. That code is using svn to manually checkout a repo that would normally be cloned with git using the go tools, so I might need to do some plumbing to make it appear to the SourceMgr as a svn repo.

Reply:
Oh, if it's not a straight case like the others, please don't bother. I didn't look at that too closely, and didn't realize it's not a svn repo... 😊

Reply:
Okay, works for me 😄 

Digging in further, I got the sense that dep might not fully support svn, in fact. It doesn't look like anything references the `svnRepo` type except one skipped test.